src/core/definitions.ts

```diff
@@ -595,12 +595,37 @@ export class GeneralOptions implements Cloneable {
 		this.themesDark = source['themesDark'];
 		this.themesDarkCustomUrl = source['themesDarkCustomUrl'];
 	}
+
+	Equals(other: GeneralOptions): Boolean {
```

Comment:
This method is not gonna work well. Settings could come from different locations like sync, backups and etc and might miss some properties.
This method should treat null/undefined and empty values as same

Reply:
with `neq` function it should now